src/plugins/intel_cpu/src/nodes/interpolate.cpp

```diff
@@ -4080,6 +4081,112 @@ void Interpolate::InterpolateRefExecutor::pillowRef(const uint8_t* in_ptr_,
     });
 }
 
+void Interpolate::InterpolateRefExecutor::pillowRefNCHWAsNHWC(const uint8_t* in_ptr_,
+                                                              uint8_t* out_ptr_,
+                                                              int B,
+                                                              int C,
+                                                              int IH,
+                                                              int IW,
+                                                              int OH,
+                                                              int OW) {
+    size_t offset = 0;
+    int filterLenX = auxTable[offset];
+    int filterLenY = auxTable[offset + 1];
+    offset += 2;
+    auto* weightX = reinterpret_cast<float*>(&auxTable[offset]);
+    offset += filterLenX * OW;
+    auto* weightY = reinterpret_cast<float*>(&auxTable[offset]);
+    offset += filterLenY * OH;
+    auto* indexX = static_cast<int*>(&auxTable[offset]);
+    offset += 2 * OW;
+    auto* indexY = static_cast<int*>(&auxTable[offset]);
+
+    bool xPass = IW != OW;
+    bool yPass = IH != OH;
+
+    auto bc_loop = [&](size_t b) {
+        const uint8_t* in_ptr_b = in_ptr_ + b * IH * IW * C * srcDataSize;
+        uint8_t* out_ptr_b = out_ptr_ + b * OH * OW * C * dstDataSize;
+
+        uint8_t* xpass_out_ptr_b = nullptr;
+        const uint8_t* ypass_in_ptr_b = nullptr;
+
+        if (xPass && yPass) {
+            size_t parallel_num = B;
+            if (parallel_num < m_threads_num) {
+                xpass_out_ptr_b = static_cast<uint8_t*>(&pillow_working_buf[b * IH * OW * C * srcDataSize]);
+                ypass_in_ptr_b = static_cast<const uint8_t*>(&pillow_working_buf[b * IH * OW * C * srcDataSize]);
+            } else {
+                size_t threadsIdx = parallel_get_thread_num();
+                auto buffer_size = static_cast<size_t>(IH) * OW * C;
+                xpass_out_ptr_b = static_cast<uint8_t*>(&pillow_working_buf[threadsIdx * buffer_size * srcDataSize]);
+                ypass_in_ptr_b =
+                    static_cast<const uint8_t*>(&pillow_working_buf[threadsIdx * buffer_size * srcDataSize]);
+            }
+        } else if (xPass && !yPass) {
+            xpass_out_ptr_b = out_ptr_b;
+        } else if (!xPass && yPass) {
+            ypass_in_ptr_b = in_ptr_b;
+        } else if (!xPass && !yPass) {
+            cpu_memcpy(out_ptr_b, in_ptr_b, OH * OW * C * dstDataSize);
```

Comment:
This code is going to be run in a single thread (within a parallel for loop), but `cpu_memcpy` does a multithread copy itself. Therefore, is not it better to use `std::memcpy` here, to avoid nested parallelism?